app/kuma-dp/pkg/dataplane/metrics/server.go

```diff
@@ -30,6 +31,23 @@ var (
 var (
 	prometheusRequestHeaders = []string{"accept", "accept-encoding", "user-agent", "x-prometheus-scrape-timeout-seconds"}
 	logger                   = core.Log.WithName("metrics-hijacker")
+
+	// Prometheus scraper supports multiple content types. Lower the number, higher the priority.
+	prometheusContentTypePriorities = map[string]int{
```

Comment:
Kind of nitpicky but IMO the other way around is _slightly_ more maintainable/clear, where we define the slice `prometheusPriorityContentType: []expfmt.Format{FmtOpenMetrics....}` to define the priorities and create a reverse lookup from the slice.

Reply:
Okay... You want to define the `[]expfmt.Format{}` sorted as per their priorities and then build reverse lookup map from it, which is nothing but `map[expfmt.Format]int{}`.
Yes, that would be nice and intuitive. I will update.